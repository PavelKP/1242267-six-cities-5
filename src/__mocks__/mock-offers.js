export const offers = [
  {
    "city": {
      "name": `Cologne`,
      "coordinates": [
        50.938361,
        6.959974
      ],
      "zoom": 13
    },
    "images": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`
    ],
    "title": `Nice, cozy, warm big bed apartment`,
    "rating": 2.5,
    "bedrooms": 2,
    "price": 485,
    "host": {
      "avatar": `img/avatar-angelina.jpg`,
      "id": 25,
      "isPro": true,
      "name": `Angelina`
    },
    "description": [
      `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`
    ],
    "id": 1,
    "isBookmarked": false,
    "isPremium": false,
    "entire": `house`,
    "adults": 6,
    "inside": [
      `Breakfast`,
      `Washer`,
      `Baby seat`,
      `Laptop friendly workspace`,
      `Air conditioning`
    ],
    "coordinates": [
      50.945361,
      6.935974
    ],
    "zoom": 16,
    "preview": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
  }
];

export const rawOffers = [
  {
    "city": {
      "name": `Cologne`,
      "location": {
        "latitude": 50.938361,
        "longitude": 6.959974,
        "zoom": 13
      }
    },
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    "images": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`
    ],
    "title": `Nice, cozy, warm big bed apartment`,
    "is_favorite": false,
    "is_premium": false,
    "rating": 2.5,
    "type": `house`,
    "bedrooms": 2,
    "max_adults": 6,
    "price": 485,
    "goods": [
      `Breakfast`,
      `Washer`,
      `Baby seat`,
      `Laptop friendly workspace`,
      `Air conditioning`
    ],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "is_pro": true,
      "avatar_url": `img/avatar-angelina.jpg`
    },
    "description": `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    "location": {
      "latitude": 50.945361,
      "longitude": 6.935974,
      "zoom": 16
    },
    "id": 1
  }
];
