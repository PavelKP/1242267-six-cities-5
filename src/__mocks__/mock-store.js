
const DEFAULT_SORTING = `popular`;

export const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

const city = {
  name: `Moscow`,
  coordinates: [111, 222],
  zoom: 12,
};

const cities = [city];

const offers = [
  {
    city: {
      name: `Moscow`,
      coordinates: [
        50.938361,
        6.959974
      ],
      zoom: 13
    },
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
    ],
    title: `The house among olive `,
    rating: 3.8,
    bedrooms: 3,
    price: 493,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    description: [
      `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`
    ],
    id: 1,
    isBookmarked: true,
    isPremium: false,
    entire: `house`,
    adults: 4,
    inside: [
      `Laptop friendly workspace`,
      `Washer`,
      `Towels`,
      `Fridge`,
      `Air conditioning`,
      `Baby seat`,
      `Breakfast`
    ],
    coordinates: [
      50.957361,
      6.9509739999999995
    ],
    zoom: 16,
    preview: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
  }
];

export default {
  DATA: {
    offers,
    cities,
  },
  INTERFACE: {
    city,
    activeSorting: DEFAULT_SORTING,
  },
  USER: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    userData: {
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
      email: `test@test.ru`,
    }
  }
};
