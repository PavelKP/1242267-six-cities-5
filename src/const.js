export const CardType = {
  MAIN: `main`,
  NEARBY: `nearby`,
  FAVORITES: `favorites`
};
