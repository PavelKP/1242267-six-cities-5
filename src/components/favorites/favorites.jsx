import React from 'react';
import {Link} from 'react-router-dom';
import OfferList from '../offer-list/offer-list';
import Header from '../header/header';
import FavoritesEmpty from '../favorites-empty/favorites-empty';
import {offersPropTypes} from '../../prop-types/prop-types';
import {CardType} from '../../const';
import {withFavoritesLoading} from '../../hocs/with-favorites-loading/with-favorites-loading';


const Favorites = ({offers, cities, loading}) => {

  if (loading) {
    return <h3>Loading...please wait</h3>;
  }

  const locations = new Set();
  offers.slice().forEach((offer) => locations.add(offer.city.name));

  const orderedLocations = cities.filter((city) => {
    return ([...locations].includes(city.name));
  });

  const isEmpty = offers.length === 0;

  return (
    <div className="page">
      <Header/>

      {isEmpty
        ? <FavoritesEmpty />
        : <React.Fragment>
          <main className="page__main page__main--favorites">
            <div className="page__favorites-container container">
              <section className="favorites">
                <h1 className="favorites__title">Saved listing</h1>
                <ul className="favorites__list">
                  {orderedLocations.map((city, i) => (
                    <li className="favorites__locations-items" key={`favorites-location-${i}`}>
                      <div className="favorites__locations locations locations--current">
                        <div className="locations__item">
                          <a className="locations__item-link" href="#">
                            <span>{city.name}</span>
                          </a>
                        </div>
                      </div>
                      <div className="favorites__places">
                        <OfferList type={CardType.FAVORITES} offers={offers.filter((offer) => offer.city.name === city.name)} />
                      </div>
                    </li>
                  ))}

                </ul>
              </section>
            </div>
          </main>
        </React.Fragment>
      }

      <footer className="footer container">
        <Link to="/" className="footer__logo-link" href="main.html">
          <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33" />
        </Link>
      </footer>
    </div>
  );
};

Favorites.propTypes = offersPropTypes;

export {Favorites};
export default withFavoritesLoading(Favorites);
