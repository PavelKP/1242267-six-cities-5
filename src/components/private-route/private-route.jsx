import React from 'react';
import {connect} from 'react-redux';
import {AuthorizationStatus} from '../../const';
import {Redirect, Route} from 'react-router-dom';
import PropTypes from 'prop-types';
import {AppRoute} from '../../const';

const PrivateRoute = ({exact, path, renderFavorites, authorizationStatus}) => {

  return (
    <Route
      path={path}
      exact={exact}
      render={(serviceProp) => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? renderFavorites(serviceProp)
            : <Redirect to={AppRoute.LOGIN}/>
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  renderFavorites: PropTypes.func.isRequired,
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
