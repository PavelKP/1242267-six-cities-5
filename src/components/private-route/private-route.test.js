import React from 'react';
import renderer from 'react-test-renderer';
import {PrivateRoute} from './private-route';
import {Favorites} from '../../components/favorites/favorites';
import configureStore from 'redux-mock-store';
import {offers} from '../../__mocks__/mock-offers';
import {Provider} from 'react-redux';
import mockStoreData, {serviceProp} from '../../__mocks__/mock-store';
import {Router} from 'react-router-dom';
import browserHistory from '../../browser-history';
import {AppRoute, AuthorizationStatus} from '../../const';

const mockStore = configureStore([]);

describe(`<PrivateRoute /> render`, () => {

  let store;

  beforeEach(() => {
    store = mockStore(mockStoreData);
  });

  it(`Should PrivateRoute render correctly`, ()=> {
    const tree = renderer.create(
        <Provider store={store}>
          <Router history={browserHistory}>
            <PrivateRoute
              exact={true}
              path={AppRoute.FAVORITES}
              renderFavorites={()=>(
                <Favorites offers={offers} serviceProp={serviceProp}/>
              )}
              authorizationStatus={AuthorizationStatus.AUTH}
            />
          </Router>
        </Provider>)
        .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
