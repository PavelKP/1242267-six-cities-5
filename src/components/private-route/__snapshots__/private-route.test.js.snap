// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> render Should PrivateRoute render correctly 1`] = `null`;
